test/ratelimit_test.cpp

```diff
@@ -345,3 +350,45 @@ TEST_F(RateLimitManagerTest, TestGetAllRulesJSON) {
     EXPECT_EQ(rules.at(0).at("entity_type").is_string(), true);
     EXPECT_EQ(rules.at(0).at("api_keys").is_array(), true);
 }
+
+TEST_F(RateLimitManagerTest, TestAutoBan) {
```

Comment:
Not related to this test: do we have a test where we multiple rules that all match (e.g. both `allow` and `deny`) and ensure that the correct order is picked?